File: api/ServerManager.Rest/Dto/ServerInfo.cs
using ServerManager.Rest.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ServerManager.Rest.Dto
{
    public class ServerInfo
    {
        public int ServerId { get; set; }
        public string Name { get; set; }
        public string Version { get; set; }
        public string Description { get; set; }
        public ServerStatus Status { get; set; }
        public ServerPropertyList Properties { get; set; }

        internal string GetUniqueServerName()
            => $"{ServerId}_{Name}_{Version}";
    }

    public enum ServerStatus
    {
        Stopped,
        Started
    }
}

File: ui/management-console/src/models/ServerInfo.ts
import { ServerProperty } from "./ServerProperty";

export class ServerInfo {
	public constructor() {
		this.serverId = 0;
		this.name = "";
		this.version = "";
		this.description = "";
		this.status = ServerStatus.Stopped;
		this.properties = [];
	}

	serverId: number;
	name: string;
	version: string;
	description: string;
	status: ServerStatus;
	properties: ServerProperty[];
}

export enum ServerStatus
{
	Stopped,
	Started
}